modules/in_band_dtmf/in_band_dtmf.c

```diff
@@ -0,0 +1,323 @@
+/**
+ * @file in_band_dtmf.c  DTMF decoder
+ */
+#include <time.h>
+#include <ctype.h>
+#include <string.h>
+#include <re.h>
+#include <rem.h>
+#include <baresip.h>
+
+
+/**
+ * @defgroup in_band_dtmf in_band_dtmf
+ *
+ * Audio filter that de- and encodes DTMF tones.
+ *
+ * The tone length of an encoded tone is 0.1s.
+ * New tones can be added while encoding is active.
+ */
+
+struct in_band_dtmf_filt_dec {
+	struct aufilt_dec_st af;  /* inheritance */
+	struct dtmf_dec *dec;
+	const struct audio *au;
+	struct le le_priv;
+	struct tmr tmr_dtmf_end;
+};
+
+
+struct in_band_dtmf_filt_enc {
+	struct aufilt_enc_st af;  /* inheritance */
+	struct mbuf *mb;
+	unsigned srate;
+	struct le le_priv;
+};
+
+
+static struct list encs;
+static struct list decs;
+
+
+static void dtmfend_handler(void *arg)
+{
```

Comment:
Please use `arg` to pass the state: `struct in_band_dtmf_filt_dec`

The current digit can be saved on the state and read here.

Reply:
fixed by https://github.com/baresip/baresip/commit/72d9c297ce45b8bfece361889761fd05919be46b